pkg/kubelet/container/runtime.go

```diff
@@ -465,6 +467,68 @@ type RuntimeStatus struct {
 	Conditions []RuntimeCondition
 }
 
+// RuntimeConfigInfo contains runtime's configuration details, eg: user-namespaces mapping between host and container
+type RuntimeConfigInfo struct {
+	UserNamespaceConfig UserNamespaceConfigInfo
+}
+
+// UserNamespaceConfigInfo contains runtime's user-namespace configuration
+type UserNamespaceConfigInfo struct {
+	UidMappings []*UserNSMapping
+	GidMappings []*UserNSMapping
+}
+
+// UserNSMaping represents mapping of user-namespaces between host and container
+type UserNSMapping struct {
+	ContainerID uint32
+	HostID      uint32
+	Size        uint32
+}
+
+// IsUserNamespaceEnabled returns true if user-namespace feature is enabled at runtime
+func (c *RuntimeConfigInfo) IsUserNamespaceEnabled() bool {
+	if len(c.UserNamespaceConfig.UidMappings) == 0 {
+		return false
+	}
+	if len(c.UserNamespaceConfig.UidMappings) == 1 &&
+		c.UserNamespaceConfig.UidMappings[0].HostID == uint32(0) && c.UserNamespaceConfig.UidMappings[0].Size == uint32(4294967295) {
```

Comment:
I think this would be more clear.
```suggestion
	if len(c.UserNamespaceConfig.UidMappings) == 1 &&
		c.UserNamespaceConfig.UidMappings[0].HostID == c.UserNamespaceConfig.UidMappings[0].ContainerID {
```